src/interaction.js

```diff
@@ -0,0 +1,84 @@
+import {distanceBetweenPoints} from 'chart.js/helpers';
+
+const interaction = {
+  modes: {
+    /**
+     * Point mode returns all elements that hit test based on the event position
+     * @param {Object} state - the state of the plugin
+     * @param {ChartEvent} event - the event we are find things at
+     * @return {Element[]} - elements that are found
+     */
+    point(state, event) {
+      return getIntersectItems(state, event);
+    },
+
+    /**
+     * Nearest mode returns the element closest to the event position
+     * @param {Object} state - the state of the plugin
+     * @param {ChartEvent} event - the event we are find things at
+     * @param {options} options - interaction options to use
+     * @return {Element[]} - elements that are found (only 1 element)
+     */
+    nearest(state, event, options) {
+      return getNearestItem(state, event, options);
+    },
+  }
+};
+
+/**
+ * Returns all elements that hit test based on the event position
+ * @param {Object} state - the state of the plugin
+ * @param {ChartEvent} event - the event we are find things at
+ * @param {options} options - interaction options to use
+ * @return {Element[]} - elements that are found
+ */
+export function getElements(state, event, options) {
+  const mode = interaction.modes[options.mode] || interaction.modes.nearest;
+  return mode(state, event, options);
+}
+
+function getIntersectItems(state, event) {
+  return state.visibleElements.filter((element) => element.inRange(event.x, event.y, true));
+}
+
+function inRangeByAxis(element, event, axis) {
+  if (axis === 'x') {
+    return element.inXRange(event.x, event.y, true);
+  } else if (axis === 'y') {
+    return element.inYRange(event.x, event.y, true);
+  }
+  return element.inXRange(event.x, event.y, true) || element.inYRange(event.x, event.y, true);
+}
```

Comment:
I'd favor a single `inRange` function, by maybe adding the axis as parameter or setting unused coordinate to undefined.
Disclaimer: I did not check the inRange logics on how complicated it would be to ignore the coordinate that is undefined.

Reply:
Yes, I arrived to the same conclusion. Only 1 method, passing axis. About the to use only inRange, well, it could be. For sure we will add some complexity but I didn't do it because I was thinking that inRange should be a method of CHARTJS Element class, which is extended here but the signature should be the same (apologize my java approach). Adding axis as last argument (with a default) it could be done

Reply:
everything in `inRange`